File: Java/src/Twin.java
import java.util.stream.Collectors;

public class Twin {
    public static void main(String[] args) {
        String a = "Marion";
        String b = "Romain";
        boolean result = isTwin(a, b);
        System.out.println(result);
    }

    static boolean isTwin(String a, String b) {
        if (a.length() != b.length()) {
            return false;
        }
        String sortedA = a.toLowerCase().chars()
                .sorted()
                .mapToObj(c -> String.valueOf((char) c))
                .collect(Collectors.joining());

        String sortedB = b.toLowerCase().chars()
                .sorted()
                .mapToObj(c -> String.valueOf((char) c))
                .collect(Collectors.joining());
                
        return sortedA.equals(sortedB);
    }



}

File: Java/kotlin/Twin.kt
package kotlin

object Twin {
    @JvmStatic
    fun main(args: Array<String>) {
        val a = "Marion"
        val b = "Romain"
        val result = isTwin(a, b)
        println(result)
    }

    fun isTwin(a: String, b: String): Boolean {
        if (a.length != b.length) {
            return false
        }
        val sortedA = a.lowercase().toCharArray().sorted().joinToString("")
        val sortedB = b.lowercase().toCharArray().sorted().joinToString("")
        return sortedA == sortedB
    }
}

